src/accordionpane/AccordionPane.ts

```diff
@@ -0,0 +1,71 @@
+import { after } from '@dojo/core/aspect';
+import { assign } from '@dojo/core/lang';
+import { DNode, WNode } from '@dojo/widget-core/interfaces';
+import { includes } from '@dojo/shim/array';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { v } from '@dojo/widget-core/d';
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+
+import TitlePane from '../titlepane/TitlePane';
+
+import * as css from './styles/accordionPane.m.css';
+
+/**
+ * @type AccordionPaneProperties
+ *
+ * Properties that can be set on AccordionPane components
+ *
+ * @property onRequestClose   Called when the title of an open pane is clicked
+ * @property onRequestOpen    Called when the title of a closed pane is clicked
+ * @property openKeys         Array of TitlePane keys indicating which panes should be open
+ */
+export interface AccordionPaneProperties extends ThemeableProperties {
+	onRequestClose?(key: string): void;
+	onRequestOpen?(key: string): void;
+	openKeys?: string[];
+};
+
+export const AccordionPaneBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class AccordionPane extends AccordionPaneBase<AccordionPaneProperties, WNode<TitlePane>> {
+	private _assignCallback(child: WNode<TitlePane>, functionName: string, callback: (key: string) => void) {
+		const existingProperty = (child.properties as any)[functionName];
+		const property = () => { callback.call(this, `${ child.properties.key }`); };
+
+		return existingProperty ? after(existingProperty, property) : property;
+	}
```

Comment:
If we update TitlePane to pass back its key in `onRequest*`, then this function isn't necessary. I think I'd prefer that update, especially since it's currently passing back `this`, which I don't think it should.

Reply:
This function is still necessary. Its purpose isn't to pass back `key` vs. a TitlePane instance, its purpose is to preserve and aspect an existing `onRequest*` method that's defined on a child TitlePane.

Reply:
Ah I see, I didn't know you wanted to allow people to add individual titlepane `onRequest*` methods in Accordion, but that makes sense